Alternative proposal to BEP038 (#1856)
bids-standard/bids-specification: src/derivatives/atlas.md

      ├─ tpl-MNI152NLin2009cAsym_res-1_label-GM_probseg.nii.gz
      ├─ tpl-MNI152NLin2009cAsym_res-1_label-WM_probseg.nii.gz
      ├─ tpl-MNI152NLin2009cAsym_res-1_T1w.nii.gz
      ├─ tpl-MNI152NLin2009cAsym_res-1_T1w.json
      ├─ tpl-MNI152NLin2009cAsym_res-2_label-brain_mask.nii.gz
      ├─ tpl-MNI152NLin2009cAsym_res-2_label-eye_mask.nii.gz
      ├─ tpl-MNI152NLin2009cAsym_res-2_label-face_mask.nii.gz
      ├─ tpl-MNI152NLin2009cAsym_res-2_label-head_mask.nii.gz
      ├─ tpl-MNI152NLin2009cAsym_res-2_label-CSF_probseg.nii.gz
      ├─ tpl-MNI152NLin2009cAsym_res-2_label-GM_probseg.nii.gz
      ├─ tpl-MNI152NLin2009cAsym_res-2_label-WM_probseg.nii.gz
      ├─ tpl-MNI152NLin2009cAsym_res-2_T1w.nii.gz
      └─ tpl-MNI152NLin2009cAsym_res-2_T1w.json
```

Since multi-subject templates and atlas involve the spatial normalization of
subjects by means of image registration processes, it is RECOMMENDED to store
the resulting transforms for each of the subjects employed to create the
output.
Please note that the specification for spatial transforms (BEP 014) is currently
under development, and therefore, the specification of transforms files may

> change in the future.

Author: @PeerHerholz this is the mention.

I didn't want to explicitly get down the `from-` / `to-` issue because that has great potential to change (or establish some extra rules so that it is unambiguous).

# Templates and atlases

In the following, we describe how the outcomes of analyses that derive from or
produce [templates and atlases](../common-principles.md#definitions) are organized
as BIDS-Derivatives.
These outcomes typically involve quantitative maps, feature maps, parcellations,
segmentations, and other knowledge annotations such as landmarks defined with
respect to individual or template brains that are supported by spaces
such as surfaces and regular grids (images).

For derivatives with atlases in their provenance corresponding to individual subjects,
the organization follows the standards for BIDS raw and derivatives.
The following entities MAY be employed to specify template- and atlas-derived results:

-    [`space-<label>`](../glossary.md#space-entities) is REQUIRED to disambiguate derivatives defined with
     respect to different [coordinate systems](../appendices/coordinate-systems.md), following the general
     BIDS-Derivatives specifications.

> -    [`cohort-<label>`](../glossary.md#cohort-entities) is REQUIRED to disambiguate derivatives defined with

Reviewer: This might already do it, but make sure that `cohort-<label>` MUST NOT be used if neither `tpl` nor `space` are used.

Reviewer 2: Is this a logical conditional that should be validated by the spec? The text above this makes it embedded within tpl:

```Text
<pipeline_name>/
    tpl-<label>/
        [cohort-<label>/]
           [<datatype>/]
               tpl-<label>_<source_entities>[_cohort-<label>][_atlas-<label>][seg-<label>][_scale-<label>][_res-<label>][_den-<label>][_desc-<label>]_<suffix>.<extension>
```

Author: Addressed with https://github.com/bids-standard/bids-specification/pull/1856/commits/c77df1048b6e17805cc5ba9d19db91286b77e88b

     when the atlas has more than one 'brain unit' resolutions, typically relating to the area covered
     by regions.

The general filename pattern for subject derivatives with templates and atlases in their provenance
follows the general BIDS-Derivatives pattern:

```Text
<pipeline_name>/
    sub-<label>/
        <datatype>/
            <source_entities>[_space-<space>][_cohort-<label>][_atlas-<label>][seg-<label>][_scale-<label>][_res-<label>][_den-<label>][_desc-<label>]_<suffix>.<extension>
```

[`atlas-<label>`](../glossary.md#atlas-entities), [`seg-<label>`](../glossary.md#segmentation-entities),
and [`scale-<label>`](../glossary.md#scale-entities) are discussed later in section
[Filenames of derivatives with atlases in their provenance](#filenames-of-derivatives-with-atlases-in-their-provenance).

For derivatives of template- and altas-generating pipelines, which typically aggregate
several sessions and/or subjects, the derivatives-specific
[`tpl-<label>` entity](../glossary.md#template-entities) is dual in terms of usage to BIDS raw's
[`sub-<label>`](../glossary.md#subject-entities), and MAY be employed as follows:

```Text
<pipeline_name>/

>     tpl-<label>/
>         [cohort-<label>/]
>            [<datatype>/]
>                <source_entities>[_cohort-<label>][_space-<space>][_atlas-<label>][seg-<label>][_scale-<label>][_res-<label>][_den-<label>][_desc-<label>]_<suffix>.<extension>

Reviewer: We should explicitly state (and add a schema check) that `tpl` and `space` are mutually exclusive.

```suggestion
    tpl-<label>/
        [cohort-<label>/]
           [<datatype>/]
               tpl-<label>_<source_entities>[_cohort-<label>][_atlas-<label>][seg-<label>][_scale-<label>][_res-<label>][_den-<label>][_desc-<label>]_<suffix>.<extension>
```

Author: Should we also remove `<source_entities>` here?

```suggestion
    tpl-<label>/
        [cohort-<label>/]
           [<datatype>/]
               tpl-<label>[_cohort-<label>][_space-<space>][_atlas-<label>][seg-<label>][_scale-<label>][_res-<label>][_den-<label>][_desc-<label>]_<suffix>.<extension>
```

Reviewer: If the original data is from a functional localizer, would you need `task-<label>`, or are we sufficiently removed that information should go in some other entity?

         "anat": {
            "sub-01_hemi-L_atlas-Schaefer2018_seg-7n_scale-100_den-164k_dseg.label.gii": "",
            "sub-01_hemi-L_atlas-Schaefer2018_seg-7n_scale-200_den-164k_dseg.label.gii": "",
            "sub-01_hemi-L_atlas-Schaefer2018_seg-7n_scale-300_den-164k_dseg.label.gii": "",
            "sub-01_hemi-L_atlas-Schaefer2018_seg-17n_scale-100_den-164k_dseg.label.gii": "",
            "sub-01_hemi-L_atlas-Schaefer2018_seg-17n_scale-200_den-164k_dseg.label.gii": "",
            "sub-01_hemi-L_atlas-Schaefer2018_seg-17n_scale-300_den-164k_dseg.label.gii": "",
            "sub-01_hemi-L_atlas-Schaefer2018_seg-kong17n_scale-100_den-164k_dseg.label.gii": "",
            "sub-01_hemi-L_atlas-Schaefer2018_seg-kong17n_scale-200_den-164k_dseg.label.gii": "",
            "sub-01_hemi-L_atlas-Schaefer2018_seg-kong17n_scale-300_den-164k_dseg.label.gii": "",
            "...": "",
            "sub-01_hemi-R_atlas-Schaefer2018_seg-kong17n_scale-300_den-164k_dseg.label.gii": "",
         },
         "bold": {
            "sub-01_task-rest_hemi-L_den-164k_bold.func.gii": "",
            "sub-01_task-rest_hemi-R_den-164k_bold.func.gii": "",
         }
      },
   }
})
}}

> ## Tabular data

Reviewer: This section needs to be merged into the bottom of common imaging derivatives.

> The `[probseg|dseg|mask|channels].tsv` file indexes and labels each node/parcel/region within the atlas.

Reviewer: Let's remove channels.tsv from this context. It may behave similarly, but it doesn't fit the "atlas" concept being defined here.

This file will be essential for downstream workflows that generate matrices or other derived files within which node/parcel/region information is required,
as the index/label fields will be used to reference the original anatomy the index/labels are derived from.
Additional fields can be added with their respective definition/description in the sidecar json file.

<table>
  <tr>
   <td>index (or placeholder in fragment in reference)
   </td>
   <td>REQUIRED. (Integer) The number associated with the node/parcel/region (right/left hemispheres may be different).
   </td>
  </tr>
  <tr>
   <td>label
   </td>
   <td>RECOMMENDED. The node name
   </td>
  </tr>
  <tr>
   <td>network_id
   </td>
   <td>OPTIONAL. Network ID the node/parcel belongs to
   </td>
  </tr>
  <tr>
   <td>network_label
   </td>
   <td>OPTIONAL. Label of Network (for example, Dorsal Attention Network)
   </td>
  </tr>
  <tr>
   <td>coordinate_report_strategy
   </td>
   <td>OPTIONAL (RECOMMENDED if x, y, z keys are specified).  The strategy used to assess and report x, y and z coordinates of a given node/parcel/region. For example, "CenterOfMass".

>    </td>

Reviewer: This column should be removed because it's redundant with the `_<probseg|dseg|mask>.json` `CoordinateReportStrategy`.

   </td>
  </tr>
  <tr>
   <td>y
   </td>
   <td>OPTIONAL. The y-coordinate of the node in the spatial reference space (See SpatialReference in the .json file)
   </td>
  </tr>
  <tr>
   <td>z
   </td>
   <td>OPTIONAL. The z-coordinate of the node in the spatial reference space (See SpatialReference in the .json file)
   </td>
  </tr>
  <tr>
   <td>hemisphere
   </td>
   <td>OPTIONAL. MUST BE ONE OF: "left", "right", "bilateral". Indicate whether the node/parcel/region is in the left or right hemispheres, or is available bilaterally.
   </td>
  </tr>
  <tr>
   <td>color
   </td>
   <td>OPTIONAL. RGB color to use for the node.
   </td>
  </tr>
  <tr>
   <td>seed
   </td>
   <td>OPTIONAL. Seed vertex/channel of the node/region
   </td>
  </tr>
  <tr>
   <td>region
   </td>
   <td>OPTIONAL. "XY", where X can be L:left, R:right, B:bilateral, and Y can be F:frontal, T:temporal, P:parietal, O:occipital
   </td>
  </tr>
</table>

Example:
```Text
index	label	network_label	hemisphere
1	Heschl's Gyrus	Somatomotor	left
2	Heschl's Gyrus	Somatomotor	right
```

## Atlas metadata

> The `[probseg|dseg|mask].json` file provides metadata to uniquely identify, describe and characterize the atlas, as well as give proper attribution to the creators.

Author: Should we replace the sidecar concept with the description file?
```suggestion
The `atlas-<label>_description.json` file provides metadata to uniquely identify, describe and characterize the atlas, as well as give proper attribution to the creators.
```

Additionally, SpatialReference serves the important purpose of unambiguously identifying the space the atlas is labeled in.

<table>
  <tr>
   <td>Name
   </td>
   <td>REQUIRED. Name of the atlas
   </td>
  </tr>
  <tr>
   <td>Description
   </td>
   <td>RECOMMENDED. Longform description of the atlas
   </td>
  </tr>
  <tr>
   <td>SpatialReference
   </td>
   <td>RECOMMENDED. Point to an existing atlas in a template space (URL or relative file path where this file is located).
   </td>
  </tr>
  <tr>
   <td>Resolution
   </td>
   <td>RECOMMENDED. Resolution atlas is provided in.
   </td>

>   </tr>

Reviewer: These are defined on derivative files already.

  <tr>
   <td>Dimensions
   </td>
   <td>RECOMMENDED. Dimensions of the atlas, MUST be 3 (for deterministic atlases) or 4 (for probabilistic atlases).
   </td>

>   </tr>

Reviewer: Definitely don't need this

  <tr>
   <td>4thDimension
   </td>
   <td>OPTIONAL. RECOMMENDED if probabilistic atlas. Should indicate what the 4th dimension entails/refers to. MUST be "Indices" or   .

>    </td>

Reviewer: I think we don't need this?

  </tr>
  <tr>
   <td>CoordinateReportStrategy
   </td>
   <td>OPTIONAL. MUST BE ONE OF: "peak", "center_of_mass", "other". Indicate the method of coordinate reporting in statistically significant clusters. Could be the "peak" statistical coordinate in the cluster or the "center_of_mass" of the cluster. RECOMMENDED if x, y ,z values are set in the .tsv file.

>    </td>

Reviewer: This should be metadata on a TSV file, not on an atlas.

})
}}

**Producing a new template AND atlas.**
Atlasing is often performed with reference to a *custom* standard space.
In this case, a feature template map is generated from all the participant(s)

> in the study, and the atlas' artifacts are produced with reference to that

Reviewer 3: It's not clear what artifacts means? Should it be label instead?
Could also just say "and the atlas is produced with reference..."

Author: Responded below, because it can mean several things (segmentations, parcellations, landmarks, connected landmarks, fiber bundle geometries, etc.), I used the term *artifacts*.

Happy to find a good alternative if it is confusing (e.g., with imaging *artifacts*).

template.

Either by generating the template space with aligning to a pre-existing template,
or by estimating a transform between templates by means of image registration,
a new template definition MUST be employed if the new template generates
a new [*space*](../common-principles.md#definitions).
For example, let's imagine that PS13 first generated a template nuclear imaging
map and after that, a corresponding [*atlas*](../common-principles.md#definitions)
was defined.
In that case, the [`atlas-<label>`](../glossary.md#atlas-entities) SHOULD be omitted

> except several atlases need specification:

Reviewer 3: So the argument here is that "_atlas-PS13" should not appear alongside "_tpl-PS13" in the example below? This would only be the case if we have several instantiations aka templates of the same atlas.

Author: Correct

<!-- This block generates a file tree.
A guide for using macros can be found at
 https://github.com/bids-standard/bids-specification/blob/master/macros_doc.md
-->
{{ MACROS___make_filetree_example({
   "ps13-pipeline": {
      "tpl-PS13": {
         "pet": {
            "tpl-PS13_desc-nopvc_dseg.nii.gz": "",
            "tpl-PS13_desc-pvc_dseg.nii.gz": "",
            "tpl-PS13_dseg.json": "",
            "tpl-PS13_dseg.tsv": "",

>             "tpl-PS13_stat-std_desc-fnirtNopvc_mimap.json": "",

Reviewer 3: We do realize that "_mimap" has not been introduced (waiting on the PET derivatives). So maybe we should _pet for now everywhere instead?

Author: I'd be much 👍 on that. With your thumbs-up, I'll go and change it :)

Author: Done in https://github.com/bids-standard/bids-specification/pull/1856/commits/947f2798039c0ab5b2d6b20c30732275204bc394

>             "tpl-PS13_stat-std_desc-fnirtNopvc_mimap.nii.gz": "",

Reviewer 3: In reality this would rarely happen without the definition of "_space-fsaverage", aka an anatomical reference space. So maybe adding  "_space-" entities in these examples to make this more realistic? (We do understand that it is possible, but is not what most people would do.)
Also in order to keep the two templates for the different spaces in the same folder, we need to have "_space-fsaverage" and  "_space-MNI152Lin" to disambiguate.

Author: Responded below -- if I understand correctly, I think this is a duplicate comment right? https://github.com/bids-standard/bids-specification/pull/1856#discussion_r1794116590

            "tpl-PS13_stat-std_desc-fnirtPvc_mimap.json": "",
            "tpl-PS13_stat-std_desc-fnirtPvc_mimap.nii.gz": "",
            "tpl-PS13_stat-std_desc-nopvc_mimap.json": "",
            "tpl-PS13_stat-std_desc-nopvc_mimap.nii.gz": "",
            "tpl-PS13_stat-std_desc-pvc_mimap.json": "",
            "tpl-PS13_stat-std_desc-pvc_mimap.nii.gz": "",
         },
      },
   }
})
}}

Let's complete the above example by adding two new atlases to the existing

> template and (*default*) atlas:

Reviewer 3: We think it's better to stick to real use cases such as Schaeffer and Kong as in your other examples to streamline everything.

Author: Resolving in favor of https://github.com/bids-standard/bids-specification/pull/1856#discussion_r1793444766 where this was first expressed.

<!-- This block generates a file tree.
A guide for using macros can be found at
 https://github.com/bids-standard/bids-specification/blob/master/macros_doc.md
-->
{{ MACROS___make_filetree_example({
   "ps13-with-atlases-pipeline": {
      "tpl-PS13": {
         "pet": {

>             "tpl-PS13_atlas-Economo1916_desc-nopvc_dseg.nii.gz": "",

Reviewer 3: In which space is this? We are guessing this is in the PS13 space, but for generic users this might be hard to guess.
So for consistency it would be good to add space here everywhere as well (fsaverage and MNI152Lin).

Author: You are guessing what the example intends to express. Since the template *instantiates* the space, then space is not necessary anymore. Think of subject:

`sub-01_atlas-Economo1916_seg-thalamus_dseg.nii.gz` - tells you that this subject has been segmented using the Economo atlas, and this file contains discrete segment (or segments) corresponding to the Thalamus. It would not make sense to say:

`sub-01_atlas-Economo1916_space-01_seg-thalamus_dseg.nii.gz`, because regardless of what other space that atlas was brought in, it is now in the subject's native space.

Now, your second sentence makes me doubt whether that was the question:

> So for consistency it would be good to add space here everywhere as well (fsaverage and MNI152Lin).

In this case, fsaverage and MNI152NLin are intendedly removed (they are shown above anyway). Here, the idea was to show the case that a PS13 template was generated in a custom space (very much revolving around @pwighton's comment about whether you want your results to be *generative* or not of a new space, in this case showing the other part when you do want to generate a new space).

So `tpl-PS13_atlas-Economo1916_space-{fsaverage,MNI152Lin}_desc-nopvc_dseg.nii.gz` would mean a very different thing.

Also, for transparency and reproducibility, I would recommend people instead (or in addition) provide the transformations between the *new-space-generating* template and fsaverage and MNI152Lin.

I'm not sure adding those conversions would help clarify this example -- I'd prefer to keep it focused on what it is trying to show.

            "tpl-PS13_atlas-Economo1916_desc-pvc_dseg.nii.gz": "",
            "tpl-PS13_atlas-Economo1916_dseg.json": "",
            "tpl-PS13_atlas-Economo1916_dseg.tsv": "",
            "tpl-PS13_atlas-RamonCajal1908_desc-nopvc_dseg.nii.gz": "",
            "tpl-PS13_atlas-RamonCajal1908_desc-pvc_dseg.nii.gz": "",
            "tpl-PS13_atlas-RamonCajal1908_dseg.json": "",
            "tpl-PS13_atlas-RamonCajal1908_dseg.tsv": "",
            "tpl-PS13_desc-nopvc_dseg.nii.gz": "",
            "tpl-PS13_desc-pvc_dseg.nii.gz": "",
            "tpl-PS13_dseg.json": "",
            "tpl-PS13_dseg.tsv": "",
            "tpl-PS13_stat-std_desc-fnirtNopvc_mimap.json": "",
            "tpl-PS13_stat-std_desc-fnirtNopvc_mimap.nii.gz": "",
            "tpl-PS13_stat-std_desc-fnirtPvc_mimap.json": "",
            "tpl-PS13_stat-std_desc-fnirtPvc_mimap.nii.gz": "",
            "tpl-PS13_stat-std_desc-nopvc_mimap.json": "",
            "tpl-PS13_stat-std_desc-nopvc_mimap.nii.gz": "",
            "tpl-PS13_stat-std_desc-pvc_mimap.json": "",
            "tpl-PS13_stat-std_desc-pvc_mimap.nii.gz": "",
         },
      },
   }
})
}}

> where the `atlas-RamonCajal1908` and `atlas-Economo1916` hypothetically define

Reviewer 3: Why hypothetically? We don't need to say hypothetical if we use real cases as mentioned above.

Author: It's easier for me to come up with examples using hypothetical atlases, but I can see how real cases are more valuable. I'll try to find time to correct my laziness.

two different atlases (please note that, often, atlases are named after
the first author and indicating a year of a reference communication).
The original *default* or *implicit* atlas' artifacts such as

> the `tpl-PS13_desc-nopvc_dseg.nii.gz` segmentation,

Reviewer 3: Here you define segmentation as an atlas artifact. But what are other atlas artifacts? Why do we need to use "artifact" instead of directly using "segmentation"? This is confusing.

Author: Does *subproduct* or *derivative* sound better than *artifact* to you? (I would avoid *derivative* in this context because the term is loaded with the overarching BIDS-Derivatives meaning)

Author: \*\* they don't sound much better to me, TBH.

which were originally generated with the PS13 template,
MAY take an [`atlas-<label>`](../glossary.md#atlas-entities) if they
need to be differentiated from the original template and atlas dataset.
However, it is RECOMMENDED that these *default* or *implicit* atlases employed
in the *custom* space they were generated only define
[`atlas-<label>`](../glossary.md#atlas-entities)
if it is necessary to disambiguate two or more atlases.

A further example of these template-and-atlas specifications
is the *Spatially Unbiased Infratentorial Template (SUIT)*
([Diedrichsen, 2006](https://doi.org/10.1016/j.neuroimage.2006.05.056)):

<!-- This block generates a file tree.
A guide for using macros can be found at
 https://github.com/bids-standard/bids-specification/blob/master/macros_doc.md
-->
{{ MACROS___make_filetree_example({
   "suit-pipeline": {
      "tpl-SUIT": {
         "anat": {
            "CHANGES": "",
            "LICENSE": "",
            "README.md": "",
            "dataset_description.json": "",

>             "tpl-SUIT_T1w.nii.gz": "",

Reviewer 3: We know that SUIT was originally by Diedrichsen defined in MNIXXX space, but a user might want to have the space entity here again.

Author: The template is distributed with transformations to MNI152NLin2009cAsym. I don't have the details very fresh in my mind now, but that suggests it was generated in a space that can be aligned *better* to MNI.

My friend @yasseraleman is using it and indeed, he first aligns SUIT's template and then either use its transforms to move everything back to MNI or bring labels from there.

I mentioned the SUIT example precisely because, despite being *close* to MNI, it's not quite there (hence, deserving it's own space denomination).
